File: CSharp-Part-2/6. Strings and Text Processing/StringsAndTextProcessing/CorrectBrackets/CorrectBrackets.cs
// Write a program to check if in a given expression the brackets are put correctly.

using System;

class CorrectBrackets
{
    static void Main()
    {
        string inputString = Console.ReadLine();
        bool res = CheckForBrackets(inputString);
        Console.WriteLine("The given expression is {0} in terms of the brackets count.", res ? "valid" : "invalid");
    }

    static bool CheckForBrackets(string expression)
    {
        int leftCount = 0;
        int rightCount = 0;
        for (int i = 0; i < expression.Length; i++)
        {
            if (expression[i] == '(')
            {
                leftCount++;
            }
            if (expression[i] == ')')
            {
                rightCount++;
            }
        }
        if (leftCount == rightCount)
        {
            return true;
        }
        else
        {
            return false;
        }
    }
}

File: JavaScript-Fundamentals/11. Strings/correct-brackets.js
// Write a JavaScript function to check if in a given expression the brackets are put correctly.

function checkBrackets(expression) {
	var leftCount = 0,
		rightCount = 0,
		i;

	for (i = 0, len = expression.length; i < len; i += 1) {
		if (expression[i] === '(') {
			leftCount++;
		}

		if (expression[i] === ')') {
			rightCount++;
		}
	}

	return leftCount === rightCount;
}

console.log(checkBrackets('((a+b)/5-d)'));
console.log(checkBrackets(')(a+b))'));